File: Android/App/app/src/main/java/com/efalco777/crud_products/domain/product/model/Product.kt
package com.efalco777.crud_products.domain.product.model


data class Product(
    val id: Int,
    val title: String,
    val description: String,
    val category: String,
    val price: Double,

    ) {
    companion object {
        // Factory function for creating an empty or default instance
        fun empty() = Product(
            id = 0,
            title = "",
            description = "",
            category = "",
            price = 0.0
        )
    }
}

File: Flutter/App/lib/domain/product/model/product.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'product.freezed.dart';

@freezed
class Product with _$Product {
  const factory Product({
    required int id,
    required String title,
    required String description,
    required double price,
    required String category,
    required List<String> images,
  }) = _Product;


}
